### PR TITLE
[Cypress] Initial tests for Applications | CU-2p7c37r

### PR DESCRIPTION
**Pull request by Chadha93**
Signed-off-by: Gaurav Chadha <gauravchadha1676@gmail.com>

**Description**
Initial test to load an application --> rename --> save --> search on canvas
CypressApplication

This PR fixes #CU-2p7c37r

**Notes for Reviewers**
The saving of the Application is failing (commented) and needs to be fixed.

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
Task linked: [CU-2p7c37r Load Application](https://app.clickup.com/t/2p7c37r)

File: cypress/integration/e2e/application_spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      })
+  });
+
+  it("Rename and Saving Application", () => {
+    cy.get("#component-drawer-Application").should('be.visible')
+    cy.get("#design-name-textfield").type("Changed Application Name with cypress");
+    cy.intercept('/api/application').as('applicationSave')
+ 
+    // TODO - Saving Application request intercept need to be fixed.
+
```
Comment (Chadha93) [inline cypress/integration/e2e/application_spec.js:32]: @Abhishek-kumar09 it is not returning the request, any suggestion to fix it?

Comment (Abhishek-kumar09) [reply cypress/integration/e2e/application_spec.js:32]: Application has problems with renaming, you should see the red snackbar with it. That's why the interception fails.

Comment (Abhishek-kumar09) [reply cypress/integration/e2e/application_spec.js:32]: Even if the test fails, write the logic correctly, the tests should be correct with their logic.

Comment (Chadha93) [reply cypress/integration/e2e/application_spec.js:32]: Sure

**Comment by Chadha93** [comment]
@Abhishek-kumar09 
image

File: cypress/integration/e2e/application_spec.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    cy.get('[data-test-id="Search"]').type(applicationName);
+    cy.intercept("/api/application*").as("applicationSearch")
+    cy.wait("@applicationSearch")
+    cy.get("#MUIDataTableBodyRow-applications-0").should("be.visible").contains(applicationName);
+  })
+
+  it("Deploy an Application", () => {
+    cy.get("[data-cy='application-drawer']").click();
+    cy.get('[data-test-id="Search"]').type(applicationName);
+    cy.intercept("/api/application*").as("patternPost")
```
Comment (Abhishek-kumar09) [inline cypress/integration/e2e/application_spec.js:55]: patternPost? 😅

File: cypress/integration/e2e/application_spec.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  });
+
+  it("Drag All Visible component on canvas", () => {
+    cy.get(".component-drawer-svg-container[draggable='true']").each(ele => {
+      const elem = cy.get(ele);
+      elem.click();
+      elem.drag("#cy-canvas-container")
+      cy.wait(500)
+    })
+  });
```
Comment (Abhishek-kumar09) [inline cypress/integration/e2e/application_spec.js:85]: This not neeeds to be here.

File: cypress/integration/e2e/application_spec.js (hunk 2 above)
Comment (Chadha93) [reply cypress/integration/e2e/application_spec.js:55]: Oops fixing it.

File: cypress/integration/e2e/application_spec.js (hunk 3 above)
Comment (Chadha93) [reply cypress/integration/e2e/application_spec.js:85]: Gotcha

**Review by Abhishek-kumar09 — APPROVED** [review]
LGTM

**Pull request merged**